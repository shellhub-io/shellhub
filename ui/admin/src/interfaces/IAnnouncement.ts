import { IAnnouncement, IAnnouncementShort } from "@/interfaces/IAnnouncement";

export type IAdminAnnouncementShort = IAnnouncementShort;

export type IAdminAnnouncement = IAnnouncement;

export type IAdminAnnouncementRequestBody = Pick<IAnnouncement, "title" | "content">;
