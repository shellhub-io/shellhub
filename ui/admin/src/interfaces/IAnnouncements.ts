export interface IAnnouncements {
  uuid: string;
  title: string;
  date: string;
}

export interface IAnnouncement {
  uuid: string;
  title: string;
  content: string;
  date: string;
}
