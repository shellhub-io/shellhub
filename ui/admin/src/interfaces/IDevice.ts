type identity = {
  mac: string;
}

type infoDetails = {
  arch: string;
  id: string;
  platform: string;
  pretty_name: string;
  version: string;
}

type position = {
  latitude: number;
  longitude: number;
}

export interface IDevice {
  created_at: string;
  identity: identity;
  info: infoDetails;
  last_seen: string;
  name: string;
  namespace: string;
  online: boolean;
  position: position;
  public_key: string;
  remoteAddr: string;
  status: string;
  tags: Array<string>;
  tenant_id:string;
  uid: string;
}
