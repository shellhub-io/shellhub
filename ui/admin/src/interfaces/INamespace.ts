type Members = {
  id: string;
  role: "administrator" | "operator" | "observer" | "owner";
  username: string;
}

type Settings = {
  session_record: boolean;
}

type Billing = {
  active: boolean;
  current_period_end: string;
  customer_id: string;
  payment_failed: null | string | boolean;
  payment_method_id: string;
  price_id: string;
  state: string;
  sub_item_id: string;
  subscription_id: string;
}

export interface INamespace {
  billing: Billing;
  created_at: string;
  devices_count: number;
  max_devices: number;
  members: Array<Members>;
  name: string;
  owner: string;
  settings: Settings;
  tenant_id: string;
}
