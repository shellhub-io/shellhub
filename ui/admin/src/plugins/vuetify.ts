// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
