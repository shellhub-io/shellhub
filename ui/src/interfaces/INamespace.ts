import { IBilling } from "./IBilling";

export interface INamespaceMember {
  id: string;
  username: string;
  role: string;
}

export interface settings {
  session_record: boolean;
  connection_announcement: string;
}

export interface INamespace {
  name: string;
  owner: string;
  tenant_id: string;
  members: INamespaceMember[];
  settings: settings;
  max_devices: number;
  devices_count: number;
  created_at: string;
  billing: IBilling;
}

export interface INamespaceInvite {
  tenant: string;
  sig: string;
}

export interface INamespaceSettings {
  connection_announcement?: string;
  session_record?: boolean;
}
export interface INamespaceResponse {
  settings?: INamespaceSettings;
  id: string;
  tenant_id: string;
  name: string;
  user_id: string;
  email: string;
  role: "administrator" | "operator" | "observer" | "owner";
}
