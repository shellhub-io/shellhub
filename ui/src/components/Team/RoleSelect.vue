<template>
  <v-select
    v-model="selectedRole"
    :items="roles"
    label="Role"
    required
    hide-details
    data-test="role-select"
  >
    <template v-slot:item="{ props, item }">
      <v-list-item v-bind="props">
        <v-list-item-subtitle class="description-text">
          {{ item.raw.description }}
        </v-list-item-subtitle>
      </v-list-item>
    </template>
  </v-select>
</template>

<script setup lang="ts">
import { BasicRole } from "@/interfaces/INamespace";

const roles = [
  {
    title: "Administrator",
    value: "administrator",
    // eslint-disable-next-line vue/max-len
    description: "Full access to the namespace, can perform all actions except managing billing.\nThis includes user and device management, but excludes billing-related operations.",
  },
  {
    title: "Operator",
    value: "operator",
    // eslint-disable-next-line vue/max-len
    description: "Can manage and operate devices, but has limited administrative privileges.\nOperators cannot change billing or ownership settings.",
  },
  {
    title: "Observer",
    value: "observer",
    description: "Can view device details and sessions but cannot make any changes.\nObservers have read-only access to monitor activity.",
  },
];

const selectedRole = defineModel<BasicRole>({ required: true });
</script>

<style scoped lang="scss">
.description-text {
  white-space: normal;
  word-break: break-word;
  max-width: 600px;
  display: block;
}
</style>
