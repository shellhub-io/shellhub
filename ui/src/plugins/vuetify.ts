// Styles
import "@mdi/font/css/materialdesignicons.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { VFileUpload, VFileUploadItem } from "vuetify/labs/VFileUpload";
import "../styles/_variables.scss";
import { aliases as faAliases, fa } from "vuetify/iconsets/fa";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#FFFFFF",
    background: "#F5F5F5",
    tabs: "#F5F5F5",
    foreground: "#F5F5F5",
    paymentForm: "#F5F5F5",
    terminal: "#0f1526",
    "v-theme-background": "#FFFFFF",
    "v-theme-surface": "#FFFFFF",
    "v-theme-card": "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
    terminal: "#0f1526",
    "v-theme-background": "#1E2127",
    "v-theme-surface": "#1E2127",
    "v-theme-card": "#22252B",
  },
};

export default createVuetify({
  components: {
    VFileUpload,
    VFileUploadItem,
  },
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
  defaults: {
    global: {

    },
    VTextField: {
      variant: "outlined",
      density: "comfortable",
    },
    VSelect: {
      variant: "outlined",
      density: "comfortable",
    },
    VTextarea: {
      variant: "outlined",
    },
  },
  icons: {
    faAliases,
    sets: {
      fa,
    },
  },
});
