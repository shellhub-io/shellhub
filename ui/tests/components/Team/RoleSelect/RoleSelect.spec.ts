import { mount, VueWrapper } from "@vue/test-utils";
import { createVuetify } from "vuetify";
import { expect, describe, it, beforeEach } from "vitest";
import RoleSelect from "@/components/Team/RoleSelect.vue";
import { BasicRole } from "@/interfaces/INamespace";

describe("RoleSelect", () => {
  let wrapper: VueWrapper<InstanceType<typeof RoleSelect>>;
  const vuetify = createVuetify();

  beforeEach(async () => {
    wrapper = mount(RoleSelect, {
      global: {
        plugins: [vuetify],
      },
      props: {
        modelValue: "administrator" as BasicRole,
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the role select component", () => {
    expect(wrapper.find('[data-test="role-select"]').exists()).toBe(true);
  });

  it("displays the initial model value", async () => {
    const selectComponent = wrapper.findComponent({ name: "VSelect" });
    expect(selectComponent.props("modelValue")).toBe("administrator");
  });

  it("has correct values for each role", () => {
    const select = wrapper.findComponent({ name: "VSelect" });
    const items = select.props("items");

    expect(items[0]).toEqual({
      title: "Administrator",
      value: "administrator",
      description: expect.stringContaining("Full access to the namespace"),
    });

    expect(items[1]).toEqual({
      title: "Operator",
      value: "operator",
      description: expect.stringContaining("Can manage and operate devices"),
    });

    expect(items[2]).toEqual({
      title: "Observer",
      value: "observer",
      description: expect.stringContaining("Can view device details"),
    });
  });
});
