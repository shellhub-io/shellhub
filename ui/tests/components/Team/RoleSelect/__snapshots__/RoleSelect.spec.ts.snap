// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoleSelect > Renders the component 1`] = `
"<div data-v-444f7389="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-select v-select--single v-select--selected" data-test="role-select">
  <!---->
  <div class="v-input__control">
    <div class="v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--variant-filled v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-controls="v-menu-v-2">
      <div class="v-field__overlay"></div>
      <div class="v-field__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background"></div>
          <div class="v-progress-linear__buffer" style="width: 0%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" for="input-v-0" aria-hidden="false">
          <!---->Role
        </label><label class="v-label v-field-label" for="input-v-0">
          <!---->Role
        </label>
        <!---->
        <div class="v-field__input" data-no-activator="">
          <!---->
          <!---->
          <div class="v-select__selection"><span class="v-select__selection-text">Administrator<!----></span></div><input size="1" type="text" id="input-v-0" aria-describedby="input-v-0-messages" inputmode="none" aria-label="Open" title="Open" required="" value="administrator">
        </div>
        <!---->
      </div>
      <!---->
      <div class="v-field__append-inner">
        <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon" aria-hidden="true"></i>
        <!---->
      </div>
      <div class="v-field__outline">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div id="input-v-0-messages" class="v-input__details" role="alert" aria-live="polite">
    <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages">
      <!---->
    </transition-group-stub>
    <!---->
  </div>
</div>"
`;
